drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -384,7 +384,7 @@ static int _set(netdev_t *netdev, netopt_t opt, void *val, size_t len)
             /* don't set res to set netdev_ieee802154_t::pan */
             break;
         case NETOPT_CHANNEL:
-            assert(len != sizeof(uint8_t));
+            assert(len <= sizeof(uint16_t));
```

Comment:
Well, that seems to be broken for the last two years anyway, but I see your point. So, an upper layer API that uses 8 bit integers for radio channels need to upcast its parameter or what is the best solution for this?